src/metabase/driver/sql_jdbc/sync/interface.clj

```diff
@@ -72,9 +72,9 @@
   overriden because SQLite is silly and only returns column information for views if the query returns a non-zero
   number of rows.
 
-    (fallback-metadata-query :postgres \"public\" \"my_table\")
-    ;; -> [\"SELECT * FROM public.my_table WHERE 1 <> 1 LIMIT 0\"]"
-  {:added "0.37.1" :arglists '([driver schema table])}
+    (fallback-metadata-query :postgres \"my_database\" \"public\" \"my_table\")
+    ;; -> [\"SELECT * FROM my_database.public.my_table WHERE 1 <> 1 LIMIT 0\"]"
+  {:added "0.37.1" :arglists '([driver db schema table])}
```

Comment:
This is a breaking change to a multimethod that could be used by external drivers. I think those driver authors would appreciate if you extended this multimethod to have two arities instead, one with the db and one without.

Reply:
Thinking about this a bit more, this is a little dangerous because while driver authors aren't forced to implement all arities, it could be easy to make a mistake by calling the wrong arity.

Maybe a better solution is to convert this multimethod into a variadic function instead. So you'd have `[driver & table-components]` instead.

Reply:
the [single call site](https://github.com/metabase/metabase/blob/e620d24b96b645e011471fdb0232c1fb9fcdd888/src/metabase/driver/sql_jdbc/sync/describe_table.clj#L85) of this multimethod will always call it with 4 arguments, so I think any drivers that implement this multimethod will need to update it to 4-arity anw.

Reply:
You _could_ modify that call site to use `apply` to apply only non-nil table components but I don't love that either, it feels like an abuse of variadic args. Make the breaking change then

Reply:
But you should change this multimethod to explain that `db` and `table` are strings. Calling them `db-name` and `table-name` should suffice